CommunicationLayer/MessagingFramework/MotorFaults.cpp

```diff
@@ -27,67 +27,109 @@
 
 namespace
 {
-   const quint8 HARDWARE_OVER_CURRENT_MASK = 0x01;
-   const quint8 SOFTWARE_OVER_CURRENT_MASK = 0x02;
-   const quint8 DC_BUS_OVER_VOLTAGE_MASK = 0x04;
-   const quint8 BAD_MOTOR_POS_HALL_MASK = 0x08;
-   const quint8 WATCHDOG_CAUSED_LAST_RESET_MASK = 0x10;
-   const quint8 CONFIG_READ_ERROR_MASK = 0x20;
-   const quint8 RAIL_UNDER_VOLT_MASK = 0x40;
-   const quint8 DESATURATION_FAULT_MASK = 0x80;
+    // Error Flags
+   const unsigned char MOTOR_OVER_SPEED_FLAG = 0x01;
+   const unsigned char SOFTWARE_OVER_CURRENT_FLAG = 0x02;
+   const unsigned char DC_BUS_OVER_VOLTAGE_FLAG = 0x04;
+   const unsigned char BAD_MOTOR_POSITION_HALL_SEQUENCE_FLAG = 0x08;
+   const unsigned char WATCHDOG_CAUSED_LAST_RESET_FLAG = 0x10;
+   const unsigned char CONFIG_READ_ERROR_FLAG = 0x20;
+   const unsigned char RAIL_UNDER_VOLTAGE_LOCK_OUT_FLAG = 0x40;
+   const unsigned char DESATURATION_FAULT_FLAG = 0x80;
+
+   // Limit Flags
+   const unsigned char OUTPUT_VOLTAGE_PWM_LIMIT_FLAG = 0x01;
+   const unsigned char MOTOR_CURRENT_LIMIT_FLAG = 0x02;
+   const unsigned char VELOCITY_LIMIT_FLAG = 0x04;
+   const unsigned char BUS_CURRENT_LIMIT_FLAG = 0x08;
+   const unsigned char BUS_VOLTAGE_UPPER_LIMIT_FLAG = 0x10;
+   const unsigned char BUS_VOLTAGE_LOWER_LIMIT_FLAG = 0x20;
+   const unsigned char IPM_OR_MOTOR_TELEMETRY_LIMIT_FLAG = 0x40;
```

Comment:
The limit flags were previously in their own file, but following the structure of the new protocol I combined them into one. Is this what you would like or would you like them separate?